engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -0,0 +1,206 @@
+package net.jqwik.api;
+
+import com.tngtech.archunit.thirdparty.com.google.common.net.*;
+
+import net.jqwik.api.statistics.*;
+
+import org.assertj.core.api.*;
+
+import java.util.stream.*;
+
+class EMailsTests {
+
+	@Property(tries = 10)
+	void containsAtSign(@ForAll("emails") String email){
+		Assertions.assertThat(email).contains("@");
+	}
+
+	@Property
+	void validLengthBeforeAt(@ForAll("emails") String email){
+		String localPart = getLocalPartOfEmail(email);
+		Assertions.assertThat(localPart.length()).isBetween(1, 64);
+	}
+
+	@Property
+	void validLengthAfterAt(@ForAll("emails") String email){
+		String domain = getDomainOfEmail(email);
+		Assertions.assertThat(domain.length()).isBetween(1, 253);
+	}
+
+	@Property
+	void validSignsBeforeAt(@ForAll("emails") String email){
```

Comment:
This property is way too long for my taste, I suggest two things: 
1) Make it two properties. One for quoted, one for unquoted local part.
2) Extract some meaningful assertion methods like "containsOnlyCharsAllowedInLocalPart" or such.
Moreover, removing valid characters until the string is supposed to be empty is clever but not readable IMO.

Reply:
3. BTW, all the `replace` calls you make here should probably be `replaceAll` but that won't matter if go for something like `assertThat(domain.chars()).allMatch(c -> ALLOWED_DOMAIN_CHARS.contains(c));`

Reply:
3.: I think that .replace is also correct in this case

Reply:
ad 3. I probably don't understand the current code: How would multiple 'a's be removed?

Reply:
.replace("a", "") remove all "a"s

For example:
`System.out.println("aaaabcaaaadaaae".replace("a", ""));`
prints:
`bcde`

Reply:
you're right. Had the impression that replace would have semantics of replaceFirst.
Anyway, you should get rid of that whole code all the same.

Reply:
Yes, this is my next step.
You're right, the replace methods are always confusing for me, too.